radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/SystemMapKey.java

```diff
@@ -47,6 +47,13 @@ public static SystemMapKey ofValidatorData(byte typeId, byte[] mapKey) {
 		return new SystemMapKey(buf.array());
 	}
 
+	public static SystemMapKey ofResourceData(REAddr addr, byte typeId) {
```

Comment:
Argument order looks inconsistent with other methods

Reply:
+1 on consistency.

Currently, we have:
- `0x00 + (epoch_data | round_data | unclaimed_address)  =>  substate_id`: system data
- `0x02 + (validator_rake | validator_owner | validator_metadata | validator_allow_delegation)  =>  substate_id`: virtual parents
- `0x02 + (validator_rake | validator_owner | validator_metadata | validator_allow_delegation) + validator_public_key  =>  substate_id`: validator data
- `0x03 + hashed_key + (resource | resource_metadata)  =>  substate_id`: resource data

Suggest to have `address + substate_type_id + bytes`, assuming 
- `0x02+public_key` is validator address (internal)
- The validator data virtual parents are of system

In short, we have:
```
0x00 + <epoch_data | round_data | unclaimed_address | validator_xxx>  => substate id
0x02 + validator_public_key + <validator_xxx>  => substate id
0x03 + hashed_key + <resource_xxx>  => substate id
```